pkg/cmd/ssh/options.go

```diff
@@ -0,0 +1,234 @@
+/*
+SPDX-FileCopyrightText: 2021 SAP SE or an SAP affiliate company and Gardener contributors
+
+SPDX-License-Identifier: Apache-2.0
+*/
+
+package ssh
+
+import (
+	"context"
+	"crypto/rand"
+	"crypto/rsa"
+	"crypto/x509"
+	"encoding/pem"
+	"errors"
+	"fmt"
+	"io"
+	"io/ioutil"
+	"net"
+	"os"
+	"path/filepath"
+	"strings"
+	"time"
+
+	"github.com/gardener/gardenctl-v2/internal/util"
+	"github.com/gardener/gardenctl-v2/pkg/cmd/base"
+	"github.com/gardener/gardener/pkg/utils"
+
+	"github.com/spf13/cobra"
+	"golang.org/x/crypto/ssh"
+	cryptossh "golang.org/x/crypto/ssh"
+	"k8s.io/cli-runtime/pkg/genericclioptions"
+)
+
+// Options is a struct to support target command
+type Options struct {
+	base.Options
+
+	// Interactive can be used to toggle between gardenctl just
+	// providing the bastion host while keeping it alive (non-interactive),
+	// or gardenctl opening the SSH connection itself (interactive). For
+	// interactive mode, a NodeName must be specified as well.
+	Interactive bool
+
+	// NodeName is the name of the Shoot cluster node that the user wants to
+	// connect to. If this is left empty, grdenctl will only establish the
+	// bastion host, but leave it up to the user to SSH themselves.
+	NodeName string
+
+	// CIDRs is a list of IP address ranges to allowed for accessing the
+	// created Bastion host. If not given, gardenctl will attempt to
+	// auto-detect the user's IP and allow only it (i.e. use a /32 netmask).
+	CIDRs []string
+
+	// SSHPublicKeyFile is the full path to the file containing the user's
+	// public SSH key. If not given, gardenctl will create a new temporary keypair.
+	SSHPublicKeyFile string
+
+	// SSHPrivateKeyFile is the full path to the file containing the user's
+	// private SSH key. This is only set if no key was given and a temporary keypair
+	// was generated. Otherwise gardenctl relies on the user's SSH agent.
+	SSHPrivateKeyFile string
+
+	// WaitTimeout is the maximum time to wait for a bastion to become ready.
+	WaitTimeout time.Duration
+
+	// KeepBastion will control whether or not gardenctl deletes the created
+	// bastion once it exits. By default it deletes it, but we allow the user to
+	// keep it for debugging purposes.
+	KeepBastion bool
+}
+
+// NewOptions returns initialized Options
+func NewOptions(ioStreams genericclioptions.IOStreams) *Options {
+	return &Options{
+		Options: base.Options{
+			IOStreams: ioStreams,
+		},
+		Interactive: true,
+		WaitTimeout: 10 * time.Minute,
+		KeepBastion: false,
```

Comment:
Because this way _everyone_ can easily get a hold of the default config, and it's not just buried in the CLI setup code. In my mind, the "ownership" of the default values belongs to the options struct, not one of its users (the cobra command).

Reply:
allright